### ISSUE TITLE
Rule Hashing Duplicates plyara Function

### ISSUE THREAD

**Issue by malvidin** [CybercentreCanada/CCCS-Yara#25]
Looking at the differences between `generate_logic_hash` from plyara.utils and `calculate_rule_hash` from the CCCS-Yara validator_functions, I can't see a significant benefit from duplicating the capability.

Is there an issue with the plyara function that hasn't been raised there?

**Comment by cccs-gm** [comment]
The calculate_rule_hash had been originally created as an adaptation of the Florian Roth algorithm as noted in the code. When I found out about the generate_logic_hash and performed some testing there appeared be a few cases where the calculate_rule_hash generated the same hash but generate_logic_hash did not.

Unfortunately all of that testing was done on a machine I do not have access to at the moment. In attempting to recreate that testing from memory I have only found one use case where there is a difference. I will be discussing it with my team this week but I suspect there is not significant benefit as you noted.
